lib/dRep/util/kbase_obj.py

```diff
@@ -0,0 +1,249 @@
+import os
+import time
+import glob
+import re
+import subprocess
+import shutil
+import pandas as pd
+from Bio import SeqIO, SeqUtils
+from functools import reduce
+import logging
+
+import drep
+
+from .dprint import dprint
+from .config import _globals
+
+
+class BinnedContigs:
+    '''
+    DS for BinnedContigs information
+    Very mutable, not necessarily in a consistent state
+
+    Instance variables created during init:
+    * upa
+    * bins_dir
+    * name
+    * assembly_upa
+    * original_bin_name_list
+
+    Other instance variables
+    * stats
+
+    '''
+
+    created_instances = [] # loaded from KBase
+    saved_instances = [] # saved to KBase
+    
+      
+    def __init__(self, upa, get_bins_dir=None, **kwargs):
+       
+        self.created_instances.append(self)
+        self.upa = upa
+
+
+        if get_bins_dir == 'download': 
+            self._load_bins_dir()
+        elif get_bins_dir == 'local': 
+            self.bins_dir = kwargs['bins_dir']
+        else:
+            raise Exception('must specify `get_bins_dir`')
+
+        self._get_obj()
+        self.original_bin_name_list = self.get_curr_bin_name_list()
+
+
+    def _load_bins_dir(self):
+        ''''''
+        logging.info(f'Downloading files for BinnedContigs with upa: {self.upa}')
+
+        dprint('_globals', run=globals())
+
+        mguObjData = _globals.mgu.binned_contigs_to_file({
+            'input_ref': self.upa, 
+            'save_to_shock': 0
+            }) # returns dict with just bin_file_directory
+
+        self.bins_dir = mguObjData['bin_file_directory']
+    
+        dprint('os.listdir(self.bins_dir)', run={**locals(), **globals()})
+
+
+    def _get_obj(self):
+        obj = _globals.ws.get_objects2({
+            'objects': [{
+                'ref': self.upa
+                }]
+            }) # huge -- includes all the statistics
+
+        self.name = obj['data'][0]['info'][1]
+        self.assembly_upa = obj['data'][0]['data']['assembly_ref']
+
+   
+    def get_curr_bin_name_list(self):
+        bin_name_list = []
+        dprint('self.bins_dir', 'os.listdir(self.bins_dir)', run={**globals(), **locals()})
+        for bin_name in next(os.walk(self.bins_dir))[2]:
+            if not re.search(r'.*\.fasta$', bin_name):
```

Comment:
another way to get the extension is `os.path.splitext(bin_name)[1]`. This might be a little more reliable. For example, your regex would successfully match files called ".fasta"

Reply:
i was aware of this. should have commented my justification